tensorbay/client/dataset.py

```diff
@@ -270,6 +358,42 @@ def upload_segment(
         multithread_upload(segment_client.upload_data, segment_filter, jobs=jobs)
         return segment_client
 
+    def upload_dataset(
```

Comment:
I think the `upload_dataset` function should not move under `DatasetClient`.
In my opinion, `upload_dataset` should still under `GAS`.
The difference is `upload_dataset` will create a draft automatically, and upload files into that draft.
Also you can input an exist and uncommit draft number into `upload_dataset`, to upload the files to that draft rather than create a new one.